MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -47,18 +50,82 @@ class QuestionFactory: QuestionFactoryProtocol {
                 text: "Рейтинг этого фильма больше чем 6?",
                 correctAnswer: false)
     ]
-    
-    init(delegate: QuestionFactoryDelegate?) {
+    */
+    init(delegate: QuestionFactoryDelegate?, moviesLoader: MoviesLoading) {
         self.delegate = delegate
+        self.moviesLoader = moviesLoader
     }
     
+    func loadData() {
+        moviesLoader?.loadMovies { result in
```

Comment:
Здесь не критично и не приводит к ошибке, однако потенциальное проблемное место из-за пропущенного ослабления ссылки 
```suggestion
        moviesLoader?.loadMovies { [weak self] result in
```